### PR TITLE
Mothership local with clusterstatus

### PR DESCRIPTION
**Pull request by pPrecel**
Description

Changes proposed in this pull request:

- Allow users to use a custom `Cluster State` as a template to reconcile cluster

**Related issue(s)**

- https://github.com/kyma-incubator/reconciler/issues/268

### CODE REVIEW

File: cmd/mothership/local/options.go [hunk 1]
```diff
@@ -112,33 +113,66 @@ func componentsFromStrings(list []string, values []string) ([]*keb.Component, er
 	return comps, nil
 }
 
+func componentsFromClusterState(cluster cluster.State, values []string) ([]*keb.Component, error) {
```
Comment (jeremyharisch) [inline cmd/mothership/local/options.go:116]: It would be nice to have a unit test for this function

**Review by jeremyharisch — COMMENTED** [review]
It would be great to cover the options.go with unit tests, other than that it looks good

File: cmd/mothership/local/options.go [hunk 2]
```diff
@@ -112,33 +113,66 @@ func componentsFromStrings(list []string, values []string) ([]*keb.Component, er
...
+		}
+	}
+
+	components := cluster.Configuration.Components
+	for i := range components {
+		name := components[i].Component
+		if vals[name] != nil {
+			mapValue, ok := vals[name].(map[string]interface{})
+			if ok {
+				for key, value := range mapValue {
```
Comment (mjakobczyk) [inline cmd/mothership/local/options.go:131]: Second inner loop with double `if` statement may be hard to read - what about decomposing it into smaller functions?

File: cmd/mothership/local/options.go (hunk 1 above)
Comment (mjakobczyk) [inline cmd/mothership/local/options.go:116]: What does this function do? It creates components from cluster state, removes them, sorts them? Some verb is missing here

Comment (pPrecel) [reply cmd/mothership/local/options.go:116]: The name of this function is consistent with other functions with similar responsibility (in the same go file) like `componentsFromFile` and `componentsFromStrings`

Comment (pPrecel) [reply cmd/mothership/local/options.go:116]: good idea! thanks

File: cmd/mothership/local/options.go (hunk 2 above)
Comment (pPrecel) [reply cmd/mothership/local/options.go:131]: This part of the code is almost the same as the [origin](https://github.com/kyma-incubator/reconciler/pull/632/files#diff-7aa83bd90c022a0af21cae41a9a134bff0edb05889da94b4d544b2e558c0cd54L93-L102) on which I based my code. I think that consistency is more important in this context. Later we can think about improving both parts

**Review by jeremyharisch — DISMISSED** [review]

**Review by jeremyharisch — APPROVED** [review]

**Pull request merged**